sky/server/common.py

```diff
@@ -371,6 +371,10 @@ def get_api_server_status(endpoint: Optional[str] = None) -> ApiServerInfo:
 
 
 def handle_request_error(response: 'requests.Response') -> None:
+    # Keep the original HTTPError if the response code >= 400
+    response.raise_for_status()
```

Comment:
What is even the point of this function then? I thought the whole purpose of this was to print a better error message. I don't think it should be possible to get a 301 or 204 or something here.
If we want to do this, I'd rather just delete the function and replace all its usages with `response.raise_for_status()` directly.

Reply:
I think the original intention of this function is to fail fast on all non-200 status code (cc @Michaelvll for confirming). Use `response.raise_for_status()` directly does not error out on 3xx status code so it will change the original fail fast behavior. How about we keep it as is?